### PR TITLE
Remove direct use of Job in Basehandler

### PR DESCRIPTION
**Pull request by henrikstranneheim**
## Description

Remove direct use of Job in Basehandler.

### Changed

- Remove direct use of Job in Basehandler and use function to delete analysis jobs

### How to prepare for test
- [ ] ssh to hasta (depending on type of change)
- [ ] activate stage: `us`
- [ ] request trailblazer-stage on hasta: `paxa`
- [ ] install on stage:
    ```Shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_trailblazer -t trailblazer -b refactor_job -a
    ```
- [ ] ssh to clinical-db (depending on type of change)
- [ ] install on stage:
`bash /home/proj/production/servers/resources/clinical-db.scilifelab.se/update-trailblazer-ui-stage.sh [THIS-BRANCH-NAME]`

### How to test
- [ ] login to ...
- [ ] do ...

### Expected test outcome
- [ ] check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review
- [ ] tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a
- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: trailblazer/store/crud/update.py [hunk 1]
```diff
@@ -37,11 +37,9 @@ def update_analysis_jobs_from_slurm_jobs(
                 data_analysis=analysis.data_analysis, job_step=job.step
             )
 
-        for job_obj in analysis.failed_jobs:
-            job_obj.delete()
-        self.commit()
+        self.delete_analysis_jobs(analysis=analysis)
         analysis.failed_jobs = [
```
Comment (fevac) [inline trailblazer/store/crud/update.py:41]: type hint?

File: trailblazer/store/crud/update.py [hunk 2]
```diff
@@ -37,11 +37,9 @@ def update_analysis_jobs_from_slurm_jobs(
                 data_analysis=analysis.data_analysis, job_step=job.step
             )
 
-        for job_obj in analysis.failed_jobs:
-            job_obj.delete()
-        self.commit()
+        self.delete_analysis_jobs(analysis=analysis)
```
Comment (fevac) [inline trailblazer/store/crud/update.py:40]: 🌟

**Review by fevac — APPROVED** [review]
Nice!! 🌟

**Review by ChrOertlin — APPROVED** [review]
Looks good, nice to have the delete in a dedicated method!

**Comment by henrikstranneheim** [comment]
### CLI ✅ 

Screenshot 2023-08-08 at 14 46 35


### API ☑️ 

Screenshot 2023-08-08 at 14 48 52

File: trailblazer/store/crud/update.py (hunk 1 above)
Comment (henrikstranneheim) [reply trailblazer/store/crud/update.py:35]: Pycharm says:
Screenshot 2023-08-08 at 14 58 24

**Comment by henrikstranneheim** [comment]
Deployed to production:
```
Log deploy... done.
trailblazer, version 15.1.6
[henrik.stranneheim@hasta:~] [P_base] 1m17s $
````

**Pull request merged**